packages/compass-logging/src/index.ts

```diff
@@ -58,7 +61,22 @@ export function createLoggerAndTelemetry(component: string): {
     event: string,
     properties: TrackProps = {}
   ): Promise<void> => {
-    const { trackUsageStatistics } = await preferencesIpc.getPreferences();
+    // Avoid circular dependency between compass-logging and compass-preferences-model
+    // Note that this is mainly a performance optimization, since the main process
+    // telemetry code also checks this preference value, so it is safe to fall back to 'true'.
+    try {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore-error Types from the dependency may not be available in early bootstrap.
+      preferencesIpc ??= (await import('compass-preferences-model'))
+        .preferencesIpc;
+    } catch {
+      preferencesIpc ??= {
+        getPreferences() {
+          return Promise.resolve({ trackUsageStatistics: true });
+        },
+      };
+    }
```

Comment:
A nit that I think might be more important later down the road when we are exploring how to make things like logger or preferences behave more like a injected services and so totally feel free to ignore this. I wonder how do you feel about already making this part a more explicit dependency injection mechanism, allowing to somehow set this interface for the logger from outside instead of having a direct dependency here on the `compass-preferences-model`. Maybe we can have some sort of `setIsTrackingEnabled(getIsEnabled: () => Promise<boolean>)` interface exported out of the compass-logging here?

Reply:
So … that sounds like a really nice thing to do, but I’m not sure how to go about it practically speaking. What would ensure that `setIsTrackingEnabled()` is called for all instances of `compass-logging`? We could manually call it in the main process and renderer process setup, that _should_ cover it, but I’m not sure that it’s good to spread the setup logic to a fairly independent place right now …

Reply:
On a module level our build setup should make sure that only one instance of `compass-logging` is being imported in the application, so using an interface to set something in the module scope once should work I think, but I also get your point, it's not the cleanest interface and it might be hard to find a good place for setting it, so then it's something that is better left for later.

I might be making a wrong assumption here, so please tell me if I'm saying something what was discussed before or just silly, but giving logging / telemetry code another look it seems like `CompassTelemetry` in `packages/compass` is currently a place where all the telemetry is being passed through and setup initially and might be a better place to depend on `compass-preferences-model` directly in general, especially now that main electron process holds the model and not the renderer. Also being the application entrypoint, the code in `packages/compass` always feels like the best place to wire up together pieces that should stay fairly independent for me, no DI required, what do you think?

Reply:
As I mentioned I think for now we can totally even keep it as-is, especially as it seems we would want more time to think this through.

FWIW addressing the tight coupling of logging and preferences overweights relying on certain part of the application init code doing its job, I am certain we would have to change this to [make tracking pluggable](https://jira.mongodb.org/browse/COMPASS-5799) for cloud team for example. I think it's just ultimately unavoidable in an application like Compass, we already make a bunch of assumptions like that in the init code and we should rather test for this working as expected (which we already do), instead of trying to avoid it

Reply:
Yeah, let’s maybe chat about this in our 1:1 tomorrow (if you’re around for it :slightly_smiling_face:)